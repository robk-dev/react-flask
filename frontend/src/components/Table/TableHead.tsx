
import React, { FC } from 'react';
import { Data, Order } from './Table';
import { createStyles, makeStyles } from '@material-ui/core/styles';
import TableSortLabel from '@material-ui/core/TableSortLabel';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';

interface HeaderProps {
    onRequestSort: (event: React.MouseEvent<unknown>, property: keyof Data) => void;
    order: Order;
    orderBy: string;
}

interface HeadCell {
    disablePadding: boolean;
    id: keyof Data;
    label: string;
    numeric: boolean;
}

const useStyles = makeStyles(() =>
    createStyles({
        hiddenSort: {
            border: 0,
            clip: 'rect(0 0 0 0)',
            height: 1,
            margin: -1,
            overflow: 'hidden',
            padding: 0,
            position: 'absolute',
            top: 20,
            width: 1,
        }
    })
);

const headCells: HeadCell[] = [
    { id: 'name', numeric: false, disablePadding: true, label: 'Name' },
    { id: 'email', numeric: false, disablePadding: false, label: 'Email' },
    { id: 'kids', numeric: true, disablePadding: false, label: 'Number of kids' },
    { id: 'date', numeric: false, disablePadding: false, label: 'Birthday' },
];

export const TableHeader: FC<HeaderProps> = (props) => {
    const classes = useStyles();
    const { order, orderBy, onRequestSort } = props;
    const createSortHandler = (property: keyof Data) => (event: React.MouseEvent<unknown>) => {
        onRequestSort(event, property);
    };

    return (
        <TableHead>
            <TableRow>
                {headCells.map((headCell) => (
                    <TableCell
                        key={headCell.id}
                        padding='default'
                        sortDirection={orderBy === headCell.id ? order : false}
                    >
                        <TableSortLabel
                            active={orderBy === headCell.id}
                            direction={orderBy === headCell.id ? order : 'asc'}
                            onClick={createSortHandler(headCell.id)}
                        >
                            {headCell.label}
                            {orderBy === headCell.id ? (
                                <span className={classes.hiddenSort}>
                                    {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
                                </span>
                            ) : null}
                        </TableSortLabel>
                    </TableCell>
                ))}
            </TableRow>
        </TableHead>
    );
}
