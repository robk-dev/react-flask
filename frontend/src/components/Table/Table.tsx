import React, { FC, useState } from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TablePagination from '@material-ui/core/TablePagination';
import TableRow from '@material-ui/core/TableRow';
import TextField from '@material-ui/core/TextField';
import Paper from '@material-ui/core/Paper';
import { TableHeader } from './TableHead';

function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
    if (b[orderBy] < a[orderBy]) {
        return -1;
    }
    if (b[orderBy] > a[orderBy]) {
        return 1;
    }
    return 0;
}

export type Order = 'asc' | 'desc';

function getComparator<Key extends keyof any>(
    order: Order,
    orderBy: Key,
): (a: { [key in Key]: number | string }, b: { [key in Key]: number | string }) => number {
    return order === 'desc'
        ? (a, b) => descendingComparator(a, b, orderBy)
        : (a, b) => -descendingComparator(a, b, orderBy);
}

export interface Data {
    id: string;
    name: string;
    date: string;
    email: string;
    kids: number;
}

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: '100%',
        },
        paper: {
            width: '100%',
            marginBottom: theme.spacing(2),
        },
        table: {
            minWidth: 750,
        }
    }),
);

export const MyTable: FC<{ rows: Data[] }> = ({ rows }) => {
    const classes = useStyles();
    const [order, setOrder] = useState<Order>('asc');
    const [orderBy, setOrderBy] = useState<keyof Data>('id');
    const [page, setPage] = useState(0);
    const [rowsPerPage, setRowsPerPage] = useState(15);
    const [search, setSearch] = useState<string>('');

    function sort_filter(array: Data[], comparator: (a: any, b: any) => number): Data[] {
        const new_arr = array.map((el, index) => [el, index] as [Data, number]); // cloning
        const filtered = search.length > 0
            ? new_arr.filter(([el]) => el.email.indexOf(search) > -1 || el.name.indexOf(search) > -1) // filter on name or email
            : new_arr;

        filtered.sort((a, b) => {
            const order = comparator(a[0], b[0]);
            if (order !== 0) return order;
            return a[1] - b[1];
        });
        return filtered.map((el) => el[0]);
    }

    const handleSearch = (event: any) => {
        setSearch(event && event.target && event.target.value);
        sort_filter(rows, getComparator(order, orderBy));
    }

    const handleRequestSort = (event: React.MouseEvent<unknown>, property: keyof Data) => {
        const isAsc = orderBy === property && order === 'asc';
        setOrder(isAsc ? 'desc' : 'asc');
        setOrderBy(property);
    };

    const handleChangePage = (_event: unknown, newPage: number) => {
        setPage(newPage);
    };

    const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
        setRowsPerPage(parseInt(event.target.value, 10));
        setPage(0);
    };

    return (
        <div className={classes.root}>
            <Paper className={classes.paper}>
                <TableContainer>
                    <TextField
                        variant="outlined"
                        margin="normal"
                        fullWidth
                        id="search"
                        label="Search..."
                        name="search"
                        autoComplete="search"
                        onChange={handleSearch}
                    />
                    <Table
                        className={classes.table}
                        aria-labelledby="tableTitle"
                        size='medium'
                        aria-label="enhanced table"
                    >
                        <TableHeader
                            order={order}
                            orderBy={orderBy}
                            onRequestSort={handleRequestSort}
                        />
                        <TableBody>
                            {sort_filter(rows, getComparator(order, orderBy))
                                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                                .map((row, index) => {
                                    const labelId = `key-${index}`;

                                    return (
                                        <TableRow
                                            hover
                                            tabIndex={-1}
                                            key={row.id}
                                        >
                                            <TableCell component="th" id={labelId} scope="row" padding="default">
                                                {row.name}
                                            </TableCell>
                                            <TableCell >{row.email}</TableCell>
                                            <TableCell >{row.kids}</TableCell>
                                            <TableCell >{row.date ? new Date(row.date).toISOString().split('T')[0] : row.date}</TableCell>
                                        </TableRow>
                                    );
                                })}
                        </TableBody>
                    </Table>
                </TableContainer>
                <TablePagination
                    rowsPerPageOptions={[15, 50, 100]}
                    component="div"
                    count={rows.length}
                    rowsPerPage={rowsPerPage}
                    page={page}
                    onChangePage={handleChangePage}
                    onChangeRowsPerPage={handleChangeRowsPerPage}
                />
            </Paper>
        </div>
    );
}
