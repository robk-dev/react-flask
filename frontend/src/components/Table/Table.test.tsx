import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MyTable, Data } from './Table';

function createData(
    id: number | string,
    name: string,
    email: string,
    kids: number,
    date: Date,
): Data {
    if (typeof id === 'number') {
        id = id.toString();
    }
    return { id, name, email, kids, date: date.toISOString().split('T')[0] };
}

const rows = [
    createData(1, 'Adam Smith', 'adamsmith@gmail.com', 1, new Date()),
    createData(2, 'John Smith', 'johnsmith@gmail.com', 2, new Date('1975-01-01')),
    createData(3, 'Jake Smith', 'jakesmith@gmail.com', 3, new Date()),
    createData(4, 'Jack Smith', 'jacksmith@gmail.com', 4, new Date()),
    createData(5, 'Jill Smith', 'jillsmith@gmail.com', 151, new Date()),
    createData(6, 'John Smith', 'johnsmith@gmail.com', 15, new Date()),
    createData(7, 'John Smith', 'johnsmith@gmail.com', 10, new Date()),
    createData(8, 'John Smith', 'johnsmith@gmail.com', 5, new Date()),
    createData(9, 'John Smith', 'johnsmith@gmail.com', 1, new Date()),
    createData(10, 'John Smith', 'johnsmith@gmail.com', 1, new Date())
];

describe('Table Test Suite', () => {
    let container: any = null;
    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });
    afterEach(() => {
        container.remove();
        container = null;
    });
    test('Table should render without crashing', () => {
        render(<MyTable rows={rows} />);
    });

    test('Table tree should match snapshot one', () => {
        const table = renderer.create(<MyTable rows={rows} />)
        const tree = table.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
