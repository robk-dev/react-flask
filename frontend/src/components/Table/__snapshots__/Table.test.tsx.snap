// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Test Suite Table tree should match snapshot one 1`] = `
<div
  className="makeStyles-root-9"
>
  <div
    className="MuiPaper-root makeStyles-paper-10 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiTableContainer-root"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="search"
          id="search-label"
        >
          Search...
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="search"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="search"
            name="search"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-14"
            >
              <span>
                Search...
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <table
        aria-label="enhanced table"
        aria-labelledby="tableTitle"
        className="MuiTable-root makeStyles-table-11"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                Name
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                Email
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                Number of kids
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                Birthday
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
            tabIndex={-1}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              id="enhanced-table-checkbox-0"
              role="cell"
              scope="row"
            >
              Adam Smith
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              adamsmith@gmail.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              1
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-10-29
            </td>
          </tr>
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
            tabIndex={-1}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              id="enhanced-table-checkbox-1"
              role="cell"
              scope="row"
            >
              John Smith
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              johnsmith@gmail.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              1
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-10-29
            </td>
          </tr>
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
            tabIndex={-1}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              id="enhanced-table-checkbox-2"
              role="cell"
              scope="row"
            >
              John Smith
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              johnsmith@gmail.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              1975-01-01
            </td>
          </tr>
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
            tabIndex={-1}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              id="enhanced-table-checkbox-3"
              role="cell"
              scope="row"
            >
              Jake Smith
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              jakesmith@gmail.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              3
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-10-29
            </td>
          </tr>
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
            tabIndex={-1}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              id="enhanced-table-checkbox-4"
              role="cell"
              scope="row"
            >
              Jack Smith
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              jacksmith@gmail.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              4
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-10-29
            </td>
          </tr>
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
            tabIndex={-1}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              id="enhanced-table-checkbox-5"
              role="cell"
              scope="row"
            >
              Jill Smith
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              jillsmith@gmail.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              151
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-10-29
            </td>
          </tr>
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
            tabIndex={-1}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              id="enhanced-table-checkbox-6"
              role="cell"
              scope="row"
            >
              John Smith
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              johnsmith@gmail.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              15
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-10-29
            </td>
          </tr>
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
            tabIndex={-1}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              id="enhanced-table-checkbox-7"
              role="cell"
              scope="row"
            >
              John Smith
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              johnsmith@gmail.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              10
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-10-29
            </td>
          </tr>
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
            tabIndex={-1}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              id="enhanced-table-checkbox-8"
              role="cell"
              scope="row"
            >
              John Smith
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              johnsmith@gmail.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              5
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-10-29
            </td>
          </tr>
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
            tabIndex={-1}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              id="enhanced-table-checkbox-9"
              role="cell"
              scope="row"
            >
              John Smith
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              johnsmith@gmail.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              1
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-10-29
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="MuiTablePagination-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          className="MuiTablePagination-spacer"
        />
        <p
          className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          Rows per page:
        </p>
        <div
          className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            15
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            tabIndex={-1}
            value={15}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          1-10 of 10
        </p>
        <div
          className="MuiTablePagination-actions"
        >
          <button
            aria-label="Previous page"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Previous page"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Next page"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Next page"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
